tmail-backend/jmap/extensions/src/test/java/com/linagora/tmail/james/jmap/StandaloneEventAttendanceRepositoryTest.java

```diff
@@ -0,0 +1,180 @@
+
+        session = MailboxSessionUtil.create(ALICE);
+        mailbox = messageIdManagerTestSystem.createMailbox(MailboxFixture.INBOX_ALICE, session);
+    }
+
+    @Test
+    void givenAcceptedFlagIsLinkedToMailGetAttendanceStatusShouldReturnAccepted() {
+        Flags flags = new Flags("$accepted");
+        MessageId messageId = createMessage(flags);
+        assertThat(Mono.from(testee.getAttendanceStatus(mailbox.getUser(), messageId)).block())
+            .isEqualTo(AttendanceStatus.Accepted);
+    }
+
+    @Test
+    void givenRejectedFlagIsLinkedToMailGetAttendanceStatusShouldReturnDeclined() {
+        Flags flags = new Flags("$rejected");
+        MessageId messageId = createMessage(flags);
+        assertThat(Mono.from(testee.getAttendanceStatus(mailbox.getUser(), messageId)).block())
+            .isEqualTo(AttendanceStatus.Declined);
+    }
+
+    @Test
+    void givenTentativelyAcceptedFlagIsLinkedToMailGetAttendanceStatusShouldReturnTentative() {
+        Flags flags = new Flags("$tentativelyaccepted");
+        MessageId messageId = createMessage(flags);
+        assertThat(Mono.from(testee.getAttendanceStatus(mailbox.getUser(), messageId)).block())
+            .isEqualTo(AttendanceStatus.Tentative);
+    }
+
+    // It should also print a warning message
+    @Test
+    void givenMoreThanEventAttendanceFlagIsLinkedToMailGetAttendanceStatusShouldReturnAny() {
+        Flags flags = new Flags("$rejected");
+        flags.add("$accepted");
+
+        MessageId messageId = createMessage(flags);
+        assertThat(Mono.from(testee.getAttendanceStatus(mailbox.getUser(), messageId)).block())
+            .satisfiesAnyOf(
+                status -> assertThat(status).isEqualTo(AttendanceStatus.Accepted),
+                status -> assertThat(status).isEqualTo(AttendanceStatus.Declined));
+    }
+
+    @Test
+    void getAttendanceStatusShouldFallbackToNeedsActionWhenNoFlagIsLinkedToMail() {
+        Flags flags = new Flags();
+        MessageId messageId = createMessage(flags);
+
+        assertThat(Mono.from(testee.getAttendanceStatus(mailbox.getUser(), messageId)).block())
+            .isEqualTo(AttendanceStatus.NeedsAction);
+    }
+
+    @Test
+    void getAttendanceStatusShouldFallbackToNeedsActionWhenNoEventAttendanceFlagIsLinkedToMail() {
+        Flags flags = new Flags(Flags.Flag.RECENT);
+        MessageId messageId = createMessage(flags);
+
+        assertThat(Mono.from(testee.getAttendanceStatus(mailbox.getUser(), messageId)).block())
+            .isEqualTo(AttendanceStatus.NeedsAction);
+    }
+
+    @Test
+    void setAttendanceStatusShouldSetAcceptedFlag() {
+        Flags flags = new Flags();
+        MessageId messageId = createMessage(flags);
+
+        Mono.from(testee.setAttendanceStatus(mailbox.getUser(), messageId, AttendanceStatus.Accepted)).block();
+
+        Flags updatedFlags = getFlags(messageId);
+
+        assertThat(updatedFlags.contains("$accepted")).isTrue();
+    }
```

Comment:
Let's test idempotance:

```suggestion
    @Test
    void setAttendanceStatusShouldSetAcceptedFlag() {
        Flags flags = new Flags();
        MessageId messageId = createMessage(flags);

        Mono.from(testee.setAttendanceStatus(mailbox.getUser(), messageId, AttendanceStatus.Accepted)).block();
        Mono.from(testee.setAttendanceStatus(mailbox.getUser(), messageId, AttendanceStatus.Accepted)).block();

        Flags updatedFlags = getFlags(messageId);

        assertThat(updatedFlags.contains("$accepted")).isTrue();
        assertThat(updatedFlags.length).isEqualTo(1);
    }
```